src/display/node_stream.js

```diff
@@ -0,0 +1,374 @@
+/* Copyright 2012 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* globals __non_webpack_require__ */
+
+let fs = __non_webpack_require__('fs');
+let http = __non_webpack_require__('http');
+let https = __non_webpack_require__('https');
+let url = __non_webpack_require__('url');
+
+import { assert, createPromiseCapability } from '../shared/util';
+import { validateRangeRequestCapabilities } from './network_utils';
+
+class PDFNodeStream {
+  constructor(options) {
+    this.options = options;
+    this.source = options.source;
+    this.url = url.parse(this.source.url);
+    this.isHttp = this.url.protocol === 'http:' ||
+                  this.url.protocol === 'https:';
+    this.isFsUrl = !this.url.host;
+    this.httpHeaders = (this.isHttp && this.source.httpHeaders) || {};
+
+    this._fullRequest = null;
+    this._rangeRequestReaders = [];
+  }
+
+  getFullReader() {
+    assert(!this._fullRequest);
+    this._fullRequest = this.isFsUrl ?
+      new PDFNodeStreamFsFullReader(this) :
+      new PDFNodeStreamFullReader(this);
+    return this._fullRequest;
+  }
+
+  getRangeReader(start, end) {
+    let rangeReader = this.isFsUrl ?
+      new PDFNodeStreamFsRangeReader(this, start, end) :
+      new PDFNodeStreamRangeReader(this, start, end);
+    this._rangeRequestReaders.push(rangeReader);
+    return rangeReader;
+  }
+
+  cancelAllRequests(reason) {
+    if (this._fullRequest) {
+      this._fullRequest.cancel(reason);
+    }
+
+    let readers = this._rangeRequestReaders.slice(0);
+    readers.forEach(function(reader) {
+      reader.cancel(reason);
+    });
+  }
+}
+
+class BaseFullReader {
+  constructor(stream) {
+    this._url = stream.url;
+    this._done = false;
+    this._errored = false;
+    this._reason = null;
+    this.onProgress = null;
+    this._length = stream.source.length;
+    this._loaded = 0;
+
+    this._fullRequest = null;
```

Comment:
This name is a bit confusing, because you use the same name elsewhere in the file.
Furthermore, `_fullRequest` is not an accurate description of the value.

How about renaming `_fullRequest` of the *FullReader classes to `_readableStream`? This would also make the code more understandable for those who are already familiar with Node.js streams.